### PR TITLE
feat!: null safety

### PR DESCRIPTION
**Pull request by felangel**
## Status

**IN DEVELOPMENT**

## Breaking Changes

YES

## Description

- [X] feat(bloc)!: nullsafety (#1932)
  - [X] published v7.0.0-nullsafety.3
- [X] feat(flutter_bloc)!: nullsafety (#2014)
  - [X] published v7.0.0-nullsafety.3
- [X] feat(bloc_test)!: nullsafety (#2067)
  - [X] published v8.0.0-nullsafety.1 (not fully null safe due to [package:test](https://pub.dev/packages/test)) ⚠️ 
- [X] feat(hydrated_bloc)!: nullsafety (#2117)
  - [X] published v7.0.0-nullsafety.2
- [X] feat(replay_bloc)!: nullsafety (#2119)
  - [X] published v0.0.1-nullsafety.2

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [X] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [X] 🧹 Code refactor
- [x] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/bloc/test/bloc_observer.mocks.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import 'package:bloc/bloc.dart';
+import 'package:mockito/mockito.dart';
+
+class MockBlocObserver extends Mock implements BlocObserver {
```
Comment (eernstg) [inline packages/bloc/test/bloc_observer.mocks.dart:4]: You should be able to leave out the forwarding methods: The superclass is `Mock` and that class has a `noSuchMethod` which is not the one from `Object`, and this means that the language automatically induces forwarding methods with the same signature as in the interface, and those forwarding methods will call `noSuchMethod` with a suitable `Invocation` argument. So you'll get code just like the methods below for free.

File: packages/bloc/pubspec.yaml [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
-version: 6.0.3
+version: 6.1.0-nullsafety
 repository: https://github.com/felangel/bloc/tree/master/packages/bloc
 issue_tracker: https://github.com/felangel/bloc/issues
 homepage: https://github.com/felangel/bloc
 documentation: https://bloclibrary.dev
 
 environment:
-  sdk: ">=2.6.0 <3.0.0"
+  sdk: ">=2.10.0-99.0.dev <3.0.0"
```
Comment (jakemac53) [inline packages/bloc/pubspec.yaml:10]: It is safest to leave this at `<2.10.0` for now - the feature might still change and if you did accidentally publish this package in this state that limits the potential negative impact.

**Review by jakemac53 — COMMENTED** [review]
This looks good to me :+1:

File: packages/bloc/test/bloc_observer.mocks.dart (hunk 1 above)
Comment (felangel) [reply packages/bloc/test/bloc_observer.mocks.dart:4]: I tried both ways and am still unable to get it working with `Mock`. I am seeing the following error when running tests:

```sh
$ pub run --enable-experiment=non-nullable test
00:01 +4: loading test/bloc_test.dart                                                                                                                                                                                                                 ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:147:23: Warning: Operand of null-aware operation '??' has type 'int' which excludes null.
  int get hashCode => _givenHashCode ?? 0;
                      ^
../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:155:24: Warning: Operand of null-aware operation '??' has type 'String' which excludes null.
  String toString() => _givenName ?? runtimeType.toString();
                       ^
../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:159:10: Warning: Operand of null-aware operation '??' has type 'Iterable<RealCall>' which excludes null.
 - 'Iterable' is from 'dart:core'.
 - 'RealCall' is from 'package:mockito/src/mock.dart' ('../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart').
        (realCalls ?? _realCalls).map((call) => call.toString());
         ^
00:01 +4 -1: loading test/bloc_test.dart [E]                                                                                                                                                                                                          
  Failed to load "test/bloc_test.dart":
  Unable to spawn isolate: Error: Cannot run with sound null safety as one or more dependencies do not
  support null safety:
  
   - package:mockito
  
  Run 'pub outdated --mode=null-safety' to determine if versions of your
  dependencies supporting null safety are available.
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:31:11: Error: Field '_whenCall' should be initialized because its type '_WhenCall' doesn't allow null.
   - '_WhenCall' is from 'package:mockito/src/mock.dart' ('../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart').
  _WhenCall _whenCall;
            ^^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:32:12: Error: Field '_untilCall' should be initialized because its type '_UntilCall' doesn't allow null.
   - '_UntilCall' is from 'package:mockito/src/mock.dart' ('../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart').
  _UntilCall _untilCall;
             ^^^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:119:61: Error: Optional parameter 'returnValue' should have a default value because its type 'Object' doesn't allow null.
   - 'Object' is from 'dart:core'.
    dynamic noSuchMethod(Invocation invocation, [Object /*?*/ returnValue]) {
                                                              ^^^^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:157:49: Error: Optional parameter 'realCalls' should have a default value because its type 'Iterable<RealCall>' doesn't allow null.
   - 'Iterable' is from 'dart:core'.
   - 'RealCall' is from 'package:mockito/src/mock.dart' ('../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart').
    String _realCallsToString([Iterable<RealCall> realCalls]) {
                                                  ^^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:346:41: Error: Optional parameter 'name' should have a default value because its type 'String' doesn't allow null.
  T named<T extends Mock>(T mock, {String name, int hashCode}) => mock
                                          ^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:346:51: Error: Optional parameter 'hashCode' should have a default value because its type 'int' doesn't allow null.
  T named<T extends Mock>(T mock, {String name, int hashCode}) => mock
                                                    ^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:681:39: Error: Optional parameter 'named' should have a default value because its type 'String' doesn't allow null.
  Null argThat(Matcher matcher, {String named}) =>
                                        ^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:686:43: Error: Optional parameter 'named' should have a default value because its type 'String' doesn't allow null.
  Null captureThat(Matcher matcher, {String named}) =>
                                            ^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:691:43: Error: Optional parameter 'named' should have a default value because its type 'String' doesn't allow null.
  Null typed<T>(ArgMatcher matcher, {String named}) => null;
                                            ^^^^^
../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:147:23: Warning: Operand of null-aware operation '??' has type 'int' which excludes null.
  int get hashCode => _givenHashCode ?? 0;
                      ^
../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:155:24: Warning: Operand of null-aware operation '??' has type 'String' which excludes null.
  String toString() => _givenName ?? runtimeType.toString();
                       ^
../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:159:10: Warning: Operand of null-aware operation '??' has type 'Iterable<RealCall>' which excludes null.
 - 'Iterable' is from 'dart:core'.
 - 'RealCall' is from 'package:mockito/src/mock.dart' ('../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart').
        (realCalls ?? _realCalls).map((call) => call.toString());
         ^
00:03 +4 -2: loading test/cubit_test.dart [E]                                                                                                                                                                                                         
  Failed to load "test/cubit_test.dart":
  Unable to spawn isolate: Error: Cannot run with sound null safety as one or more dependencies do not
  support null safety:
  
   - package:mockito
  
  Run 'pub outdated --mode=null-safety' to determine if versions of your
  dependencies supporting null safety are available.
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:31:11: Error: Field '_whenCall' should be initialized because its type '_WhenCall' doesn't allow null.
   - '_WhenCall' is from 'package:mockito/src/mock.dart' ('../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart').
  _WhenCall _whenCall;
            ^^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:32:12: Error: Field '_untilCall' should be initialized because its type '_UntilCall' doesn't allow null.
   - '_UntilCall' is from 'package:mockito/src/mock.dart' ('../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart').
  _UntilCall _untilCall;
             ^^^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:119:61: Error: Optional parameter 'returnValue' should have a default value because its type 'Object' doesn't allow null.
   - 'Object' is from 'dart:core'.
    dynamic noSuchMethod(Invocation invocation, [Object /*?*/ returnValue]) {
                                                              ^^^^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:157:49: Error: Optional parameter 'realCalls' should have a default value because its type 'Iterable<RealCall>' doesn't allow null.
   - 'Iterable' is from 'dart:core'.
   - 'RealCall' is from 'package:mockito/src/mock.dart' ('../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart').
    String _realCallsToString([Iterable<RealCall> realCalls]) {
                                                  ^^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:346:41: Error: Optional parameter 'name' should have a default value because its type 'String' doesn't allow null.
  T named<T extends Mock>(T mock, {String name, int hashCode}) => mock
                                          ^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:346:51: Error: Optional parameter 'hashCode' should have a default value because its type 'int' doesn't allow null.
  T named<T extends Mock>(T mock, {String name, int hashCode}) => mock
                                                    ^^^^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:681:39: Error: Optional parameter 'named' should have a default value because its type 'String' doesn't allow null.
  Null argThat(Matcher matcher, {String named}) =>
                                        ^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:686:43: Error: Optional parameter 'named' should have a default value because its type 'String' doesn't allow null.
  Null captureThat(Matcher matcher, {String named}) =>
                                            ^^^^^
  ../../../../../../.pub-cache/hosted/pub.dartlang.org/mockito-4.1.2/lib/src/mock.dart:691:43: Error: Optional parameter 'named' should have a default value because its type 'String' doesn't allow null.
  Null typed<T>(ArgMatcher matcher, {String named}) => null;
                                            ^^^^^
00:03 +19 -2: Some tests failed.
```

Do you have any advice/suggestions? I tried the codegen approach but `build_runner` couldn't find any classes to generate. Is this a case where we need to resort to using `Fakes`?

Comment (eernstg) [reply packages/bloc/test/bloc_observer.mocks.dart]: First, you need to run the tests with `--no-sound-null-safety` because the test programs import `mockito` which doesn't support null safety yet:

```
> pub run --enable-experiment=non-nullable --no-sound-null-safety test
```

My comment about omitting the forwarding methods was based on properties of the language Dart&mdash;you will indeed get the noSuchMethod forwarders that I mentioned.

However, with mockito the `Mock...` class needs to use a different signature for methods taking parameters with a potentially non-nullable type; and for methods whose return type is potentially non-nullable it needs to call `super.noSuchMethod` with an extra argument. Cf. https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md.

The reason for this is that the setup (`when` and `verify`) relies on passing null as actual arguments, so the mock object methods must permit that. And when the mock object method returns it needs to return a value of the correct type; when the return type is potentially non-nullable it can't use null, and in that case the forwarder must provide some value (any value) of the correct type.

So you get something like the following:

```dart
class MockBlocObserver extends Mock implements BlocObserver {
  @override
  void onChange(Cubit? cubit, Change? change) {
    super.noSuchMethod(Invocation.method(#onChange, [cubit, change]));
  }

  @override
  void onTransition(Bloc? bloc, Transition? transition) {
    super.noSuchMethod(Invocation.method(#onTransition, [bloc, transition]));
  }

  @override
  void onEvent(Bloc? bloc, Object? event) {
    super.noSuchMethod(Invocation.method(#onEvent, [bloc, event]));
  }

  @override
  void onError(Cubit? cubit, Object? error, StackTrace? stackTrace) {
    super.noSuchMethod(Invocation.method(#onError, [cubit, error, stackTrace]));
  }
}
```

The code generation approach is also described on https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md, but manually writing the above will give the same effect.

Comment (felangel) [reply packages/bloc/test/bloc_observer.mocks.dart]: I ended up removing the dependency on `mockito` but what would be the equivalent of

```dart
verifyNever(observer.onError(any, any, any));
````

Screen Shot 2020-09-18 at 12 11 00 AM

Comment (eernstg) [reply packages/bloc/test/bloc_observer.mocks.dart]: Sounds like you migrated the mockito `Mock` class and declarations like `verifyNever`, or you wrote something which will behave in the same way for your purposes? So `verifyNever` is declared, and it relies on the log data which is similar to that of the `Mock` class to check that `onError` was indeed never called.

If you have a `MockBlockObserver` with an `onError` method which is declared in a library where null safety is enabled, you need to declare that `onError` method such that all parameter types are nullable (so they end in `?` unless it's `void` or `dynamic`, which are nullable already), and similarly for all other methods. With that, you can pass `any` to `onError`, and the logging mechanism should ensure that you get a correct check.

Comment (felangel) [reply packages/bloc/test/bloc_observer.mocks.dart]: Ah that makes sense 👍
Thanks so much for the detailed explanations, I really appreciate it!

File: packages/bloc/lib/src/bloc.dart [hunk 3]
```diff
@@ -25,9 +25,9 @@ abstract class Bloc<Event, State> extends Cubit<State>
   /// The current [BlocObserver].
   static BlocObserver observer = BlocObserver();
 
-  final _eventController = StreamController<Event>.broadcast();
+  late final _eventController = StreamController<Event>.broadcast();
```
Comment (Jomik) [inline packages/bloc/lib/src/bloc.dart:28]: `late` seems a bit redundant here, as we are immediately accessing the variable in our `_bindEventsToStates`
```suggestion
  final _eventController = StreamController<Event>.broadcast();
```

File: packages/bloc/lib/src/bloc.dart [hunk 4]
```diff
@@ -25,9 +25,9 @@ abstract class Bloc<Event, State> extends Cubit<State>
...
 
-  StreamSubscription<Transition<Event, State>> _transitionSubscription;
+  StreamSubscription<Transition<Event, State>>? _transitionSubscription;
```
Comment (Jomik) [inline packages/bloc/lib/src/bloc.dart:30]: Can this not be late? We know that we are immediately setting it in `_bindEventsToStates` in the constructor.
```suggestion
  late StreamSubscription<Transition<Event, State>> _transitionSubscription;
```

File: packages/bloc/example/main.dart [hunk 5]
```diff
@@ -22,7 +22,7 @@ class SimpleBlocObserver extends BlocObserver {
   }
 
   @override
-  void onError(Cubit cubit, Object error, StackTrace stackTrace) {
+  void onError(Cubit cubit, Object error, StackTrace? stackTrace) {
```
Comment (jakemac53) [inline packages/bloc/example/main.dart:25]: I believe that generally stack traces are guaranteed to be non-null now (at least in the core apis) - although I don't know if there are `bloc` specific apis that don't guarantee that and would have to etc? It might be worth trying to force this non-nullable and see how that shakes out.

File: packages/bloc/lib/src/cubit.dart [hunk 6]
```diff
@@ -148,12 +147,11 @@ abstract class Cubit<State> extends Stream<State> {
   /// handlers.
   @override
   StreamSubscription<State> listen(
-    void Function(State) onData, {
-    Function onError,
-    void Function() onDone,
-    bool cancelOnError,
+    void Function(State)? onData, {
```
Comment (jakemac53) [inline packages/bloc/lib/src/cubit.dart:150]: It doesn't seem right to allow null here, are you sure you need/want that?

File: packages/bloc/lib/src/cubit.dart [hunk 7]
```diff
@@ -148,12 +147,11 @@ abstract class Cubit<State> extends Stream<State> {
...
+    Function? onError,
+    void Function()? onDone,
+    bool? cancelOnError,
   }) {
-    _controller ??= StreamController<State>.broadcast();
     return _controller.stream.listen(
       onData,
```
Comment (jakemac53) [inline packages/bloc/lib/src/cubit.dart:156]: I am surprised that this isn't an error, this shouldn't be a nullable argument as far as I can tell - https://api.dart.dev/stable/2.10.0/dart-async/Stream/listen.html.

File: packages/bloc/pubspec.yaml [hunk 8]
```diff
@@ -1,20 +1,25 @@
 name: bloc
 description: A predictable state management library that helps implement the BLoC (Business Logic Component) design pattern.
-version: 6.0.3
+version: 6.1.0-nullsafety
```
Comment (jakemac53) [inline packages/bloc/pubspec.yaml:3]: This should be `7.0.0-nullsafety` to treat it as a breaking change.

File: packages/bloc/pubspec.yaml [hunk 9]
```diff
@@ -1,20 +1,25 @@
...
 issue_tracker: https://github.com/felangel/bloc/issues
 homepage: https://github.com/felangel/bloc
 documentation: https://bloclibrary.dev
 
+# Can't be published until null safety is shipped without requiring a flag
+publish_to: none
+
 environment:
-  sdk: ">=2.6.0 <3.0.0"
+  sdk: ">=2.10.0-99.0.dev <2.10.0"
```
Comment (jakemac53) [inline packages/bloc/pubspec.yaml:13]:
```suggestion
  sdk: ">=2.10.0-99.0.dev <2.11.0"
```

We now have the 2.11 dev sdk which you can safely allow.

**Review by jakemac53 — COMMENTED** [review]
Generally looks good, my main feedback would be around the nullable StackTrace arguments which I _think_ should be non-nullable.

File: packages/bloc/lib/src/cubit.dart (hunk 7 above)
Comment (felangel) [reply packages/bloc/lib/src/cubit.dart:156]: looks like it is https://github.com/dart-lang/sdk/blob/34e8b4229efec853ae171dc1569c03b1d053c58f/sdk/lib/async/stream.dart#L507 😕

File: packages/bloc/lib/src/cubit.dart (hunk 6 above)
Comment (felangel) [reply packages/bloc/lib/src/cubit.dart:150]: I was just trying to conform to the underlying API which appears to allow null (https://github.com/dart-lang/sdk/blob/34e8b4229efec853ae171dc1569c03b1d053c58f/sdk/lib/async/stream.dart#L507)

File: packages/bloc/example/main.dart (hunk 5 above)
Comment (felangel) [reply packages/bloc/example/main.dart:25]: good point 👍

File: packages/bloc/lib/src/cubit.dart (hunk 6 above)
Comment (jakemac53) [reply packages/bloc/lib/src/cubit.dart:148]: The docs here were/are out of date - we updated [handleError](https://github.com/dart-lang/sdk/blob/34e8b4229efec853ae171dc1569c03b1d053c58f/sdk/lib/async/stream.dart#L696) actually after I presumed you were likely mislead here, but it looks like `listen` also needs to be updated :).

File: packages/bloc/lib/src/cubit.dart (hunk 7 above)
Comment (jakemac53) [reply packages/bloc/lib/src/cubit.dart:154]: Ah it looks like that is because we allow you to set this lazily. Weird (and unfortunate I think...), but fair enough :).

File: packages/bloc/lib/src/cubit.dart (hunk 6 above)
Comment (jakemac53) [reply packages/bloc/lib/src/cubit.dart:148]: Sent out https://dart-review.googlesource.com/c/sdk/+/166460 to update the docs

**Review by diegoflassa — APPROVED** [review]

**Comment by felangel** [comment]
Published [bloc v7.0.0-nullsafety.0](https://pub.dev/packages/bloc/versions/7.0.0-nullsafety.0) 🎉

**Comment by felangel** [comment]
Published [flutter_bloc v7.0.0-nullsafety.0](https://pub.dev/packages/flutter_bloc/versions/7.0.0-nullsafety.0) 🎉

**Comment by felangel** [comment]
Published [bloc v7.0.0-nullsafety.1](https://pub.dev/packages/bloc/versions/7.0.0-nullsafety.1) 🎉

**Comment by felangel** [comment]
Published [flutter_bloc v7.0.0-nullsafety.1](https://pub.dev/packages/flutter_bloc/versions/7.0.0-nullsafety.1) 🎉

**Comment by felangel** [comment]
Published [hydrated_bloc v7.0.0-nullsafety.0](https://pub.dev/packages/hydrated_bloc/versions/7.0.0-nullsafety.0) 🎉

**Comment by felangel** [comment]
Published [replay_bloc v0.0.1-nullsafety.0](https://pub.dev/packages/replay_bloc) 🎉

**Comment by zepfietje** [comment]
@felangel it seems like `bloc_test` should no longer be blocked (https://github.com/dart-lang/test/pull/1371#issuecomment-774114485);

**Comment by felangel** [comment]
> @felangel it seems like `bloc_test` should no longer be blocked ([dart-lang/test#1371 (comment)](https://github.com/dart-lang/test/pull/1371#issuecomment-774114485)).

I believe it's still blocked because the test package depends on meta 1.3.0 and it is incompatible with Flutter unless you're on the master channel.

**Comment by zepfietje** [comment]
Ah alright, didn't look into transitive dependencies.

**Comment by felangel** [comment]
Published [bloc v7.0.0-nullsafety.2](https://pub.dev/packages/bloc/versions/7.0.0-nullsafety.2) 🎉

**Comment by felangel** [comment]
Published [bloc_test v8.0.0-nullsafety.0](https://pub.dev/packages/bloc_test/versions/8.0.0-nullsafety.0) 🎉 

*Still not fully null-safe because a dependency is not null-safe:

```
Because:
* `bloc_test` that depends on:
* `test` that depends on:
* `test_core` that depends on:
* `vm_service` that doesn't opt in to null safety
</details>
<details>
<summary>
For more information
</summary>
```

**Comment by felangel** [comment]
Published [flutter_bloc v7.0.0-nullsafety.2](https://pub.dev/packages/flutter_bloc/versions/7.0.0-nullsafety.2) 🎉

**Comment by felangel** [comment]
Published [hydrated_bloc v7.0.0-nullsafety.1](https://pub.dev/packages/hydrated_bloc/versions/7.0.0-nullsafety.1) 🎉

**Comment by felangel** [comment]
Published [replay_bloc v0.0.1-nullsafety.1](https://pub.dev/packages/replay_bloc/versions/0.0.1-nullsafety.1) 🎉

**Comment by felangel** [comment]
Published [bloc v7.0.0-nullsafety.3](https://pub.dev/packages/bloc/versions/7.0.0-nullsafety.3) 🎉

**Comment by felangel** [comment]
Published [bloc_test v8.0.0-nullsafety.1](https://pub.dev/packages/bloc_test/versions/8.0.0-nullsafety.1) 🎉

**Comment by felangel** [comment]
Published [flutter_bloc v7.0.0-nullsafety.3](https://pub.dev/packages/flutter_bloc/versions/7.0.0-nullsafety.3) 🎉

**Comment by felangel** [comment]
Published [hydrated_bloc v7.0.0-nullsafety.2](https://pub.dev/packages/hydrated_bloc/versions/7.0.0-nullsafety.2) 🎉

**Comment by felangel** [comment]
Published [replay_bloc v0.0.1-nullsafety.2](https://pub.dev/packages/replay_bloc/versions/0.0.1-nullsafety.2) 🎉

**Comment by felangel** [comment]
Finally time to merge this 🎉 

I'm going to hold off on publishing a stable release for a bit longer in order to get a few more issues included but should hopefully have a stable release in the next week 👍
